src/state.rs

```diff
@@ -0,0 +1,22 @@
+/// The state of a single unit
+#[derive(RustcEncodable, RustcDecodable, PartialEq)]
+pub struct Unit {
+    pub position: [u64; 2],
+    pub angle: u64,
+    pub speed_vector: [u64; 2],
```

Comment:
The problem with angle is, that it's difficult to calculate with it. You have to calculate a d_x/d_y to get a new position and this is impossible without rounding errors. Keeping only the speed_vector leaves you in the position of not having a direction if the speed is zero.
Maybe something like
```Rust
enum Vector {
  Angle(u64),
  Speed(u64,u64),
}
```
So it's either an angle while standing still or a vector with x- and y-speed.